pip/rules.bzl

```diff
@@ -288,18 +311,25 @@ deploy_pip = rule(
     attrs = {
         "target": attr.label(
             mandatory = True,
-            allow_single_file = [".tar.gz"],
             providers = [PyDeploymentInfo],
             doc = "`assemble_pip` label to be included in the package",
         ),
+        "pypi_profile": attr.string(
+            default = "",
+            doc = "name of pypirc profile to deploy to"
+        ),
         "snapshot": attr.string(
-            mandatory = True,
+            default = "",
             doc = "Remote repository to deploy pip snapshot to"
         ),
         "release": attr.string(
-            mandatory = True,
+            default = "",
             doc = "Remote repository to deploy pip release to"
         ),
+        "deploy_types": attr.string_list(
```

Comment:
Hmmm.

I think, given that we _assemble_ the wheel every time, there's no additional harm in also deploying it every time. Which would reduce the number of parameters here and simplify the deployment code.

(You may have made this change - making the deployment optional - part way through developing this PR, in which case, sorry about that!)

So yeah, I think we should always deploy the wheel.

Reply:
reverted.